### PR TITLE
(feat): Adding minimal permissions for chaos-operator

### PR DESCRIPTION
**Pull request by shubhamchaudhary**
Signed-off-by: shubhamchaudhary <shubham.chaudhary@mayadata.io>

**What this PR does / why we need it**:

- Adding minimal permission for service account used for the creation of chaos-operator

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] Labelled this PR & related issue with `documentation` tag
- [ ] PR messages has document related information
- [ ] Labelled this PR & related issue with `breaking-changes` tag
- [ ] PR messages has breaking changes related information
- [ ] Labelled this PR & related issue with `requires-upgrade` tag
- [ ] PR messages has upgrade related information
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: deploy/rbac.yaml [hunk 1]
```diff
@@ -19,9 +19,9 @@ metadata:
   labels:
     name: litmus
 rules:
-- apiGroups: ["*"]
-  resources: ["*"]
-  verbs: ["*"]
+- apiGroups: ["","apps","batch","litmuschaos.io"]
+  resources: ["pods","jobs","deployments","daemonsets","events","configmaps","services","chaosengines","chaosexperiments"]
```
Comment (ksatchit) [inline deploy/rbac.yaml]: is there any need to add finalizer resources separately ?

Comment (shubhamchaudhary) [reply deploy/rbac.yaml]: Yes, It is showing error If we are not passing these resources and  deletecollection verb

Comment (ksatchit) [reply deploy/rbac.yaml]: though we aren't using in today, can you also add chaosResults

Comment (ksatchit) [reply deploy/rbac.yaml]: do we also need removal of secrets? (they are created by users, not the experiment itself, so should be fine, right? )

Comment (shubhamchaudhary) [reply deploy/rbac.yaml]: Yes, I will add secrets also

**Pull request merged**